### PR TITLE
fix: delete pod who is terminating

### PR DESCRIPTION
**Pull request by OrangeBao**
#### What type of PR is this?

/kind bug

#### Which issue(s) does this PR fix?

Fixes #
#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: pkg/clustertree/cluster-manager/controllers/pod/root_pod_controller.go [hunk 1]
```diff
@@ -825,6 +832,9 @@ func (r *RootPodReconciler) DeletePodInLeafCluster(ctx context.Context, lr *leaf
 	if err != nil {
 		if errors.IsNotFound(err) {
 			klog.V(4).Infof("Tried to delete pod %s/%s, but it did not exist in the cluster", leafPod.Namespace, leafPod.Name)
+			if cleanflag {
+				return cleanRootPodFunc()
```
Comment (hanweisen) [inline pkg/clustertree/cluster-manager/controllers/pod/root_pod_controller.go:836]: why make cleanRootPodFunc here instead of call function directly?

Comment (OrangeBao) [reply pkg/clustertree/cluster-manager/controllers/pod/root_pod_controller.go:836]: Multiple calls to simplify statements

**Pull request merged**